app/pages/api/realms/[id].ts

```diff
@@ -217,6 +217,40 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse<
           console.error(`Error sending email for ${updatedRealm.realm}`, err),
         );
 
+        let typeOfContactUpdate = '';
+
+        if (
+          currentRequest.approved &&
+          currentRequest.status === StatusEnum.APPLIED &&
+          currentRequest.productOwnerEmail !== updateRequest.productOwnerEmail
+        ) {
+          typeOfContactUpdate = 'Product Owner';
+          onboardNewRealmAdmin(
+            session,
+            updatedRealm,
+            currentRequest.productOwnerEmail!,
+            updatedRealm.productOwnerEmail,
+            typeOfContactUpdate,
+          );
+          offboardRealmAdmin(session, updatedRealm, currentRequest.productOwnerEmail!, typeOfContactUpdate);
+        }
+
+        if (
+          currentRequest.approved &&
+          currentRequest.status === StatusEnum.APPLIED &&
+          currentRequest.technicalContactEmail !== updateRequest.technicalContactEmail
+        ) {
+          typeOfContactUpdate = 'Technical Contact';
+          onboardNewRealmAdmin(
```

Comment:
I had a strange issue earlier where if I wasn't awaiting promises, they would be flaky on whether they actually settled. I think related to this thread: https://github.com/vercel/next.js/discussions/14077, if it doesn't complete fast enough it might get skipped. It would be nice to not wait for this to resolve before sending a response but might be necessary

Reply:
@jlangy yes agreed, the `sendEmail` function does return promise so it's good to await for them before delivering the response and in this way we can ensure the emails do get delivered. I have added async awaits for all the email. Can you please take a look tomorrow morning.